MapboxNavigation/InstructionPresenter.swift

```diff
@@ -25,15 +25,18 @@ class InstructionPresenter {
         for component in instruction {
             let isFirst = component == instruction.first
             let joinChar = !isFirst ? " " : ""
+            
+            let joinCharPlusAbbreviation = component.abbreviation != nil ? joinChar + component.abbreviation! : nil
+            let joinCharPlusText = component.text != nil ? joinChar + component.text! : nil
 
             if let shieldKey = component.shieldKey() {
                 if let cachedImage = imageRepository.cachedImageForKey(shieldKey) {
                     string.append(NSAttributedString(string: joinChar))
                     string.append(attributedString(withFont: label.font, shieldImage: cachedImage))
                 } else {
                     // Display road code while shield is downloaded
-                    if let text = component.text {
-                        string.append(NSAttributedString(string: joinChar + text, attributes: attributesForLabel(label)))
+                    if let text = joinCharPlusText {
+                        string.append(NSAttributedString(string: (text).abbreviated(toFit: label.availableBounds(), font: label.font, possibleAbbreviation: joinCharPlusAbbreviation), attributes: attributesForLabel(label)))
```

Comment:
I was thinking about adding some sort of counter that tracks the current abbreviation priority level. We'd then pass it into this function and then it'd tell us if the priority level impacted whether a abbreviation was used.

Reply:
Another option is to first sort all the components, then try and fit them, then un-sort them.

Reply:
I was thinking you would need at most two nested loops, one to iterate over priority level `i` and one to iterate over components `j`. The inner loop would join strings together, using `component[j].abbreviation` if `component[j].abbreviationPriority <= i`, else using `component[j].text`. Does that make sense?